src/transformers/models/dpt/gradient_convolution.py

```diff
@@ -0,0 +1,451 @@
+        1https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html) The amount of output padding along a
+        given dimension must be lower than the stride along that same dimension. If set to *None* (default), the output
+        shape is inferred. If both *output_padding* and *output_shape* are specified, they have to be mutually
+        compatible.
+      output_shape: Output shape of the spatial dimensions of a transpose
+        convolution. Can be *None* or an iterable of *n* integers. If a *None* value is given (default), the shape is
+        automatically calculated. Similar to *output_padding*, *output_shape* is also for disambiguating the output
+        shape when stride > 1 (see also https://www.tensorflow.org/api_docs/python/tf/nn/conv2d_transpose) If both
+        *output_padding* and *output_shape* are specified, they have to be mutually compatible.
+      dilation: *None*, or a sequence of *n* integers, giving the
+        dilation factor to apply in each spatial dimension of *rhs*. Dilated convolution is also known as atrous
+        convolution.
+      dimension_numbers: tuple of dimension descriptors as in
+        lax.conv_general_dilated. Defaults to tensorflow convention.
+      transpose_kernel: if *True* flips spatial axes and swaps the input/output
+        channel axes of the kernel. This makes the output of this function identical to the gradient-derived functions
+        like keras.layers.Conv2DTranspose and torch.nn.ConvTranspose2d applied to the same kernel. Although for typical
+        use in neural nets this is unnecessary and makes input/output channel specification confusing, you need to set
+        this to *True* in order to match the behavior in many deep learning frameworks, such as TensorFlow, Keras, and
+        PyTorch.
+      precision: Optional. Either `None`, which means the default precision for
+        the backend, a `lax.Precision` enum value (`Precision.DEFAULT`, `Precision.HIGH` or `Precision.HIGHEST`) or a
+        tuple of two `lax.Precision` enums indicating precision of ``lhs``` and `rhs`.
+    Returns:
+      Transposed N-d convolution.
+    """
+    assert len(lhs.shape) == len(rhs.shape) and len(lhs.shape) >= 2
+    ndims = len(lhs.shape)
+    one = (1,) * (ndims - 2)
+    # Set dimensional layout defaults if not specified.
+    if dimension_numbers is None:
+        if ndims == 2:
+            dimension_numbers = ("NC", "IO", "NC")
+        elif ndims == 3:
+            dimension_numbers = ("NHC", "HIO", "NHC")
+        elif ndims == 4:
+            dimension_numbers = ("NHWC", "HWIO", "NHWC")
+        elif ndims == 5:
+            dimension_numbers = ("NHWDC", "HWDIO", "NHWDC")
+        else:
+            raise ValueError("No 4+ dimensional dimension_number defaults.")
+    dn = conv_dimension_numbers_(lhs.shape, rhs.shape, dimension_numbers)
+    # dn = dimension_numbers
+    k_shape = np.take(rhs.shape, dn.rhs_spec)
```

Comment:
I would advocate for using `jnp` calls instead of `np`. This function is going to be jit’d, so we want to keep everything on the accelerator device where possible. `np` calls that pull variables back to the host.

Reply:
I found this a bit tricky, moving it to `jnp` calls will implicate going through [this statement](https://github.com/younesbelkada/transformers/blob/a40f21dfe73079f26c5afccccad1e1cf14d15ac8/src/transformers/models/dpt/gradient_convolution.py#L208): 
```
if expected_input_size != input_size,:
            raise ValueError(
                "The expected input size with the current set of input "
                f"parameters is {expected_input_size} which doesn't "
                f"match the actual input size {input_size}."
            )
``` 
Which I found a bit hard to refactor to make it run under `jit`: https://jax.readthedocs.io/en/latest/errors.html#jax.errors.ConcretizationTypeError 
Do you have any idea how we can go around this statement 🙏 ?

Reply:
Could try with a `if not jnp.equal(expected_input_size, input_size):`